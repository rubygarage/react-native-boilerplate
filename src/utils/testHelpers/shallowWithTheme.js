import React from 'react';
import { ThemeProvider } from 'styled-components/native';
// eslint-disable-next-line import/no-extraneous-dependencies
import { shallow } from 'enzyme';

import mockedTheme from './mockedTheme';

export default (tree) => shallow(
  <ThemeProvider theme={mockedTheme}>
    {tree}
  </ThemeProvider>,
);
