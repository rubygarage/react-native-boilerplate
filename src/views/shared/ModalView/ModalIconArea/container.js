import { withTheme } from 'styled-components/native';

import ModalIconAreaComponent from './component';

export default withTheme(ModalIconAreaComponent);
