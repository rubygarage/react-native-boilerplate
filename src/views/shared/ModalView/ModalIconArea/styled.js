import styled from 'styled-components/native';

const Wrap = styled.View`
  justify-content: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
`;

const Icon = styled.View`
  margin-bottom: 16px;
`;

const Text = styled.Text`
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

export { Wrap, Icon, Text };
