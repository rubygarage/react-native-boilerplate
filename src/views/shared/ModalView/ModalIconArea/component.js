import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import * as S from './styled';

const ModalIconAreaComponent = ({ iconName, textId, theme }) => (
  <S.Wrap>
    <S.Icon name={iconName} size={84} color={theme.colors.grey100} />
    <S.Text>
      <FormattedMessage id={textId} />
    </S.Text>
  </S.Wrap>
);

ModalIconAreaComponent.propTypes = {
  iconName: PropTypes.string.isRequired,
  textId: PropTypes.string.isRequired,
  theme: PropTypes.shape({ colors: PropTypes.shape().isRequired }).isRequired,
};

export default ModalIconAreaComponent;
