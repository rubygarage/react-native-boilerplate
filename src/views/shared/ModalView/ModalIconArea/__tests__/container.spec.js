import React from 'react';

import shallowWithTheme from 'utils/testHelpers/shallowWithTheme';
import diveTo from 'utils/testHelpers/diveToEnzyme';
import ModalIconArea from '../container';
import ModalIconAreaComponent from '../component';

const defaultProps = { iconName: 'close', textId: 'text_id' };

describe('ModalIconArea container', () => {
  it('renders correctly default', () => {
    const wrapper = shallowWithTheme(<ModalIconArea {...defaultProps} />);
    const container = diveTo(wrapper, ModalIconAreaComponent);
    expect(container).toMatchSnapshot();
  });
});
