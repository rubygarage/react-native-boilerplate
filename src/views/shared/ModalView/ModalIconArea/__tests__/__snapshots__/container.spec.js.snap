// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalIconArea container renders correctly default 1`] = `
<Styled(View)>
  <Styled(View)
    color="[mocked grey100]"
    name="close"
    size={84}
  />
  <Styled(Text)>
    <FormattedMessage
      id="text_id"
    />
  </Styled(Text)>
</Styled(View)>
`;
